crowbar_framework/lib/crowbar/checks/network.rb

```diff
@@ -69,7 +69,7 @@ def ip_configured?
         end
 
         def ping_succeeds?
-          system("ping -c 1 #{fqdn} > /dev/null 2>&1")
+          system("ping -c 1 #{fqdn} > /dev/null 2>&1 || ping6 -c 1 #{fqdn} > /dev/null 2>&1")
```

Comment:
This goes into a similar direction to what @dirkmueller already wrote:

Shouldn't we just call `ping` vs. `ping6` depending on the return values of the `ipv4_addrs` and `ipv6_addrs` methods (see line 87 and following)? I.e. call ping6 if just ipv6_addrs is not empty, call ping if only ipv4_addrs is not empty.

(Call both if this is a dual stack setup?)

Reply:
ahh, cool, thanks Ralf. Let me look at that. Thanks, the gut's of crowbar (and ruby) is still rather new to me :)